### PR TITLE
chore: Automate releases with semantic-release

### PR DESCRIPTION
**Pull request by ikhemissi**
Use [semantic-release](https://www.npmjs.com/package/semantic-release) to automate the publishing of Buslane to [npmjs](https://www.npmjs.com/package/@5app/buslane) whenever a PR is merged into `master`.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,20 +1,61 @@
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: circleci/node:12.9.1
```
Comment (MrSwitch) [inline .circleci/config.yml]: Can probably use the latest 

```suggestion
    - image: circleci/node:12
```
... if that's a thing, no?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,20 +1,61 @@
...
-            - "node_modules"
+            - ./node_modules
+      - run:
+          name: Lint
+          command: npm run lint
       - run:
-          name: test
-          command: npm test
+          name: Test
+          command: npm run test
```
Comment (MrSwitch) [inline .circleci/config.yml]:
```suggestion
          command: npm test
```

saves a whole 4 bytes - your welcome

File: Dockerfile [hunk 3]
```diff
@@ -1,8 +1,6 @@
-FROM node:8.8.1-alpine
+FROM node:12.9.1-alpine
```
Comment (MrSwitch) [inline Dockerfile]: Same as above, do we have to lock this down?

```suggestion
FROM node:12-alpine
```

Comment (ikhemissi) [reply Dockerfile]: It depends on the use case I guess.
In this case, we can use either as this Dockerfile is only used for testing locally (when running the command `npm run dockerTest`), but in general it is better to specify a particular tag to guarantee the same behaviour between dev/qa/prod.

File: .circleci/config.yml (hunk 1 above)
Comment (ikhemissi) [reply .circleci/config.yml]: yes, that version is supported: https://circleci.com/docs/2.0/docker-image-tags.json 👍

**Comment by MrSwitch** [comment]
@ikhemissi please squash merge when ready

**Pull request merged**